sn_networking/src/event/kad.rs

```diff
@@ -402,9 +406,45 @@ impl SwarmDriver {
                     Self::send_record_after_checking_target(sender, peer_record.record, &cfg)?;
                 } else {
                     debug!("For record {pretty_key:?} task {query_id:?}, fetch completed with split record");
-                    sender
-                        .send(Err(GetRecordError::SplitRecord { result_map }))
-                        .map_err(|_| NetworkError::InternalMsgChannelDropped)?;
+
+                    // Ensure we're not expecting a concrete record.
+                    if cfg.accumulate_spend_attempts && cfg.target_record.is_none() {
+                        let mut all_found_spends = vec![];
+                        for (_record_hash, (record, _peers)) in result_map {
+                            match get_raw_signed_spends_from_record(&record) {
+                                Ok(spends_here) => {
+                                    all_found_spends.extend(spends_here);
+                                }
+                                Err(NetworkError::RecordKindMismatch(record_kind)) => {
+                                    trace!("For record {pretty_key:?} task {query_id:?}, found record with kind {record_kind:?} instead of spend record");
+                                }
+                                Err(error) => {
+                                    error!("Unexpected error when trying to accumulate spend atttempts: {error:?}");
+                                }
+                            }
+                        }
+
+                        if all_found_spends.is_empty() {
+                            warn!("For record {pretty_key:?} task {query_id:?}, found no spend records");
```

Comment:
We'd want to send something back to the `sender` here. Else the caller would be hanging on the get operation forever.

Something like:
```rust
 sender
    .send(Err(GetRecordError::NoSpendsFoundDuringAccumulation))
    .map_err(|_| NetworkError::InternalMsgChannelDropped)?;
}
```

Thinking about it, can we move the `accumulation of spend attempts` to a higher level?
We return `SplitRecord` error here, so we can use that to accumulate the spends? It'd keep the `kad get flow` clean imo?